Uplift/Views/GymDetail/FitnessCenter/GymDetailFitnessCenterHoursCell.swift

```diff
@@ -2,13 +2,238 @@
+        let view = UIView()
+        view.backgroundColor = .gray01
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        hours = OpenHours(openHours: [])
+        hoursTableView.delegate = self
+        hoursTableView.dataSource = self
+        self.contentView.addSubview(hoursTitleLabel)
+        self.contentView.addSubview(hoursTableView)
+        self.contentView.addSubview(dividerView)
+        setupConstraints()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public func configure(delegate: GymDetailFitnessCenterHoursCellDelegate, hours: OpenHours, isDisclosed: Bool) {
+        self.isDisclosed = isDisclosed
+        self.delegate = delegate
+        self.hours = hours
+        DispatchQueue.main.async {
+            // reload table view & remake constraints
+            self.hoursTableView.reloadData()
+            self.tableViewHeight?.constraint.deactivate()
+
+            self.hoursTableView.snp.makeConstraints { make in
+                self.tableViewHeight = make.height.equalTo(self.getTableViewHeight())
+            }
+            self.layoutIfNeeded()
+        }
+    }
+
+    private func setupConstraints() {
+
+        hoursTitleLabel.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalToSuperview().offset(GymDetailConstraints.verticalPadding)
+            make.height.equalTo(GymDetailConstraints.titleLabelHeight)
+        }
+
+        hoursTableView.snp.makeConstraints { make in
+            make.centerX.width.equalToSuperview()
+            make.width.equalToSuperview()
+            make.top.equalTo(hoursTitleLabel.snp.bottom).offset(LayoutConstants.hoursTableViewTopPadding)
+            tableViewHeight = make.height.equalTo(getTableViewHeight())
+        }
+
+        dividerView.snp.makeConstraints { make in
+            make.height.equalTo(GymDetailConstraints.dividerViewHeight)
+            make.leading.trailing.equalToSuperview()
+            make.bottom.equalToSuperview()
+        }
+    }
+
+    private func getTableViewHeight() -> CGFloat {
+        if hours.isEmpty() {
+            return 0.0
+        }
+
+        var height = LayoutConstants.hoursTableViewBottomRowHeight
+
+        if isDisclosed {
+         height += CGFloat(hours.getNumHoursLines() - 1) * LayoutConstants.hoursTableViewRowHeight
+        }
+
+        return height
+    }
+
+    public static func getHeight(isDisclosed: Bool, numLines: Int) -> CGFloat {
+        var height = LayoutConstants.hoursTableViewBottomRowHeight
+        if isDisclosed {
+            height += CGFloat(numLines - 1) * LayoutConstants.hoursTableViewRowHeight
+        }
+        height += GymDetailConstraints.verticalPadding
```

Comment:
Same question here, is doing the += on multiple line for code clarity?

Reply:
yup easier to read imo